zulip/zulip/__init__.py

```diff
@@ -1270,16 +1270,15 @@ def add_subscriptions(self, streams: Iterable[Dict[str, Any]], **kwargs: Any) ->
             request=request,
         )
 
-    def remove_subscriptions(self, streams: Iterable[str], principals: Optional[Iterable[str]] = None) -> Dict[str, Any]:
+    def remove_subscriptions(self, streams: Iterable[str],
+                             principals: Optional[Union[List[str], List[int]]] = None) -> Dict[str, Any]:
         '''
             See examples/unsubscribe for example usage.
         '''
-        if principals is None:
-            principals = []
```

Comment:
Why did you change this?  I think it was better with this on its own line.

Reply:
@timabbott The reason the change happened here was to work around `mypy` being finicky about shadowing a variable and thinking we've changed the type of it to `List[Any]` (figuratively).

More discussion here: https://chat.zulip.org/#narrow/stream/92-learning/topic/mypy.20error/near/902548

Are we enforcing that the Python versions >= 3.6 yet for the API?  This would influence our decision here on how best to deal with mypy.

Reply:
@timabbott I was facing an issue. I've explained it here: https://chat.zulip.org/#narrow/stream/92-learning/topic/mypy.20error

Reply:
We probably shouldn't limit this to disallow 3.5 before it reaches EOL.  Seems really cheap to keep support and I'm sure someone will have a bad experience if we do.